### DISCUSSION TITLE
Accessing the full url inside an endpoint

### DISCUSSION THREAD

**Discussion by xiyu95** [reduxjs/redux-toolkit#2834] [Q&A] [answered]
I am wondering if there is any way to access the entire url inside an endpoint? I have a custom `dynamicBaseQuery` for my `baseApi`, whereas, I split the endpoints down into separate files. But I need to get a full url string and pass down to a util function within an endpoint query function, such as below:

```ts
updateData: build.mutation<any, ArgType>({
      query: ({ name, body }) => {

        // Looking for getting this information below in a smart way
        // A way to extract the base url from baseQuery and concatenate it and the below url field?
        const fullUrl = 'http://my-sample-server/api/data/meta/test-name'

        const neededInfo = myUtilFunc(fullUrl),
        
        return {
          url: `data/meta/${name}`,
          method: 'PUT',
          body,
          headers: { 'If-Match': neededInfo }
        };
      },
    }),
```

**Comment by phryneas** [comment, marked as answer]
That's unfortunately impossible since `baseUrl` is an implementation detail of `fetchBaseQuery`, so it's not available outside of that.

This could be more possible in `prepareHeaders`, but I think you won't be able to access the url there as well. If not that, you'd have to provide your own `fetchFn`, I'm sorry.

**Reply by xiyu95** [reply]
I see, thanks for the answer, I will need to look for some other solution. Maybe for now I could pass that in as a param.
